azure-devops/azext_devops/dev/team/service_endpoint.py

```diff
@@ -92,52 +84,86 @@ def create_service_endpoint(service_endpoint_type, authorization_scheme, name,
-            'creationMode': 'Manual'
-        }
-        service_endpoint_to_create = ServiceEndpoint(
-            authorization=service_endpoint_authorization, data=service_endpoint_data,
-            name=name, type=SERVICE_ENDPOINT_TYPE_AZURE_RM, url='https://management.azure.com/')
+        service_endpoint_authorization.parameters['authenticationType'] = 'spnKey'
+        service_endpoint_authorization.parameters['serviceprincipalkey'] = azure_rm_service_principal_key
+    else:
+        with open(azure_rm_service_principal_certificate_path, "r") as f:
+            service_endpoint_authorization.parameters['authenticationType'] = 'spnCertificate'
+            service_endpoint_authorization.parameters['servicePrincipalCertificate'] = f.read()
+
+    service_endpoint_data = {
+        'subscriptionId': azure_rm_subscription_id,
+        'subscriptionName': azure_rm_subscription_name,
+        'environment': 'AzureCloud',
+        'creationMode': 'Manual'
+    }
+    service_endpoint_to_create = ServiceEndpoint(
+        authorization=service_endpoint_authorization, data=service_endpoint_data,
+        name=name, type=SERVICE_ENDPOINT_TYPE_AZURE_RM, url='https://management.azure.com/')
+    return client.create_service_endpoint(service_endpoint_to_create, project)
+
+
+def create_github_service_endpoint(name, github_url,
+                                   organization=None, project=None, detect=None):
+    """ Create a GitHub service endpoint.
+    :param name: Name of service endpoint to create
+    :type name: str
+    :param github_url: Url for github for creating service endpoint
+    :type github_url: str
+    :rtype: :class:`ServiceEndpoint <service_endpoint.v4_1.models.ServiceEndpoint>`
+    """
+    organization, project = resolve_instance_and_project(detect=detect,
+                                                         organization=organization,
+                                                         project=project)
+    client = get_service_endpoint_client(organization)
+    if AZ_DEVOPS_GITHUB_PAT_ENVKEY not in os.environ:
+        error_message = 'Please pass GitHub access token in ' + AZ_DEVOPS_GITHUB_PAT_ENVKEY +\
+                        ' environment variable in non-interactive mode.'
+        verify_is_a_tty_or_raise_error(error_message)
+        github_access_token = prompt_pass('GitHub access token:', confirm=True)
+    else:
+        github_access_token = os.environ[AZ_DEVOPS_GITHUB_PAT_ENVKEY]
+
+    service_endpoint_authorization = EndpointAuthorization(
+        parameters={'accessToken': github_access_token},
+        scheme=SERVICE_ENDPOINT_AUTHORIZATION_PERSONAL_ACCESS_TOKEN)
+    service_endpoint_to_create = ServiceEndpoint(
+        authorization=service_endpoint_authorization,
+        name=name, type=SERVICE_ENDPOINT_TYPE_GITHUB, url=github_url)
+    return client.create_service_endpoint(service_endpoint_to_create, project)
+
+
+def create_service_endpoint(service_endpoint_configuration, organization=None,
```

Comment:
This will need some help doc or sample on how to form the configuration file..

Reply:
agree, will work on this with @geverghe